pkg/client/events/replay_client.go

```diff
@@ -228,6 +219,19 @@ func (rClient *replayClient[T, R]) retryPublishEventsFactory(ctx context.Context
 			rClient.newMapEventsBytesToTFn(errCh),
 		)
 
+		// Subscribe to the eventBzObs and block until the channel closes.
+		// Then pass this as an error to force the retry.OnError to resubscribe.
+		go func() {
+			eventsBzSub := eventsBz.Subscribe(ctx)
+			for range eventsBzSub.Ch() {
+				// Wait for the channel to close.
+				continue
+			}
+			typedObs.UnsubscribeAll()
```

Comment:
@bryanchriswhite Is this the idiomatic pattern we're following for unsubscription?

Reply:
I'm not sure I understand the question as it was intended but this looks correct to me.